calico-vpp-agent/cni/network_vpp.go

```diff
@@ -210,16 +210,23 @@ func (s *Server) AddVppInterface(podSpec *storage.LocalPodSpec, doHostSideConf b
 			goto err
 		}
 	} else {
+		pblswIfIndex, _ := podSpec.GetParamsForIfType(podSpec.PortFilteredIfType)
 		swIfIndex, isL3 = podSpec.GetParamsForIfType(podSpec.DefaultIfType)
 		if swIfIndex != types.InvalidID {
 			s.log.Infof("pod(add) Default routes to swIfIndex=%d isL3=%t", swIfIndex, isL3)
-			err = s.RoutePodInterface(podSpec, stack, swIfIndex, isL3)
+			err = s.RoutePodInterface(podSpec, stack, swIfIndex, isL3, pblswIfIndex != types.InvalidID)
```

Comment:
Instead of a `inPodVRF` it might be better to have two functions :
- `AddRouteNetworkVRFtoPodInterface()`
- `AddRoutePodVRFtoPodInterface()`